misk-hibernate-testing/src/main/kotlin/misk/jdbc/MySqlTestDatabasePoolBackend.kt

```diff
@@ -0,0 +1,70 @@
+package misk.jdbc
+
+import com.zaxxer.hikari.util.DriverDataSource
+import misk.environment.Environment
+import java.sql.Connection
+import java.sql.SQLException
+import java.util.Properties
+import javax.inject.Inject
+import javax.inject.Singleton
+import javax.sql.DataSource
+
+/**
+ * Implementation of [TestDatabasePool.Backend] for SQL databases.
+ */
+@Singleton
+class MySqlTestDatabasePoolBackend @Inject constructor(
+  val config: DataSourceConfig
+) : TestDatabasePool.Backend {
+  private val connection: Connection
+
+  init {
+    connection = try {
```

Comment:
note to self- can we defer the connect call until later?

Reply:
yes! made it a `lazy` property instead. Building a database connection is an expensive enough operation to warrant the overhead of the delegate.